features/FxA-15-OAuth-Management/README.md

```diff
@@ -0,0 +1,83 @@
+Disconnect OAuth services from FxA
+======================
+
+https://mozilla.aha.io/features/FXA-15
+
+## Problem Summary
+
+As a Firefox Accounts user I want to be able to revoke access to the
+services that I logged into.  
+
+****
+
+## Outcomes
+
+The success criteria of this feature:
+* Being able to view logged in OAuth services in the apps view.
+* Being able to revoke access ("disconnect") a service.
+
+See "Detailed Design" below for more details.
+
+## Hypothesis
+
+We believe that by adding OAuth services and functionality to disconnect the services
+for Firefox Accounts users will improve the quality and security of FxA
+for everybody.
+We will know this is true when we see activity in the "Connected apps and devices"
+
+## Metrics
+
+We would collect usage metrics with StatsD:
+* number of views of the apps and devices view,
+* number of revoke actions for any service.
+* number of revoke actions for each service by client id.
+
+Derive a percentage of settings views vs apps view to understand
+the popularity of this feature.
+
+****
+
+## Detailed design
+
+<img src='pr_moz_mock.png' width='450' />
+
+The initial OAuth apps list would
+revoke the non-expired tokens using the web UI.
+It will be displayed to logged in users of FxA
+in the "Connected apps and devices" settings view.
+This new functionality changes the current devices view and adds to it.
+
+The list will also include the sessions that were created
+for the Firefox Accounts website and the Firefox Desktop browser.
+This gives users full control over all of their sessions.
```

Comment:
I feel like there may be some interesting technical details here that we should call out.  Will we need to ask the auth-server for a list of all sessionTokens, display those that have device records as devices, and display the leftover ones as "Firefox Accounts website"?  We can't necessarily rely on the oauth server to give us a complete list here.